### PR TITLE
`Plagiarism checks`: Improve email subject for new plagiarism notification

### PR DESCRIPTION
**Pull request by rriyaldhi**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I implemented the changes with a good performance and prevented too many database calls.
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

This PR is a follow-up of [#6174](https://github.com/ls1intum/Artemis/pull/6174). The email subject of the new plagiarism case does not provide any information regarding the exercise and course in which the plagiarism case happened. To make it clear to the student, the new plagiarism case email subject should provide more information about the case.

### Description

Add course name and exercise name to the new plagiarism case email subject

### Steps for Testing

**For a detailed example of testing, please see this screencast:**
https://user-images.githubusercontent.com/10885503/217796578-1e0c1940-e80c-4b5d-9c7a-e17c43ec5eee.mov

Prerequisites:
- Test Server 0
- 1 Programming Exercise with the same code submissions to trigger the plagiarism (e.g. [Test 6215](https://artemis-test0.artemis.in.tum.de/courses/5/exercises/80))
- 1 Student with your email

1. Log in as the user with your email
2. Navigate to Exercise [Test 6215](https://artemis-test0.artemis.in.tum.de/courses/5/exercises/80)
3. Click Start Exercise
4. Click Open code editor, then click Open-graded code editor
5. Click BubbleSort.java
6. Insert bubble sort code inside the performSort method: 
```
for (int i = input.size() - 1; i >= 0; i--) {
    for (int j = 0; j < i; j++) {
        if (input.get(j).compareTo(input.get(j + 1)) > 0) {
            Date temp = input.get(j);
            input.set(j, input.get(j + 1));
            input.set(j + 1, temp);
        }
    }
}
```
7. Click Submit
8. Log in to Artemis as an Instructor
9. Navigate to Exercise [Test 6215](https://artemis-test0.artemis.in.tum.de/course-management/5/programming-exercises/80/plagiarism)
10. Click **Rerun plagiarism detection**
11. Click the result, then click Confirm
12. Click the course name of the breadcrumb navigation
13. Click Plagiarism Cases
14. Scroll to Exercise Test 6215
15. Click the case of the account with your email
16. Click Notify Student
17. Click Save
18. Check your email and verify that the email subject contains the course & exercise name

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| MailService.java | 93% | 98% |

### Screenshots

**Before**
image

**After**
image

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/notifications/NotificationSettingsService.java [hunk 1]
```diff
@@ -127,9 +127,9 @@ public class NotificationSettingsService {
 
     // This set has to equal the UI configuration in the client notification settings structure file!
```
Comment (JohannesStoehr) [inline src/main/java/de/tum/in/www1/artemis/service/notifications/NotificationSettingsService.java:128]: This comment states, that you also have to change things in the client, but I cannot see any according changes

**Review by danielbode — APPROVED** [review]
Tested on ts0. Received an email with the new title.

**Review by YeetTheFirst21 — DISMISSED** [review]
Tested during Testing Session on ts0 and seems to be working properly

Screenshot 2023-02-08 164354

**Review by frederik-kt — DISMISSED** [review]
Received an email with the new subject, looks good.

File: src/main/resources/i18n/messages.properties [hunk 2]
```diff
@@ -94,6 +94,7 @@ email.notification.aux.difficulty.medium=Medium
 email.notification.aux.difficulty.hard=Hard
 
 # Plagiarism
+email.plagiarism.title=Exercise {0} in the course {1}: New Plagiarism Case
```
Comment (ge65cer) [inline src/main/resources/i18n/messages.properties:97]: I'd prefer`New Plagiarism Case: Exercise "{0}" in the course "{1}"` but that is a matter of taste

File: src/main/java/de/tum/in/www1/artemis/service/notifications/NotificationSettingsService.java (hunk 1 above)
Comment (rriyaldhi) [reply src/main/java/de/tum/in/www1/artemis/service/notifications/NotificationSettingsService.java:128]: I removed the constant

File: src/main/resources/i18n/messages.properties (hunk 2 above)
Comment (rriyaldhi) [reply src/main/resources/i18n/messages.properties]: Changed the title according to your suggestion. It looks better. Thanks!

**Review by JohannesStoehr — APPROVED** [review]
Comments addressed and code looks good to me

**Review by Vinicius-Agreste — APPROVED** [review]
Tested on ts1 worked as expected. Successfully got my first plagiarism email ever!

**Review by YeetTheFirst21 — APPROVED** [review]
Tesed on TS 1 at the test session, looks good.

Screenshot 2023-02-08 164354

**Review by pellumbbaboci — DISMISSED** [review]
Tested on ts1 during testing session, works as expected.

**Review by krusche — CHANGES_REQUESTED** [review]
I am not happy with some of the design changes, because you violate best practices for object oriented programming (in particular you do not achieve low coupling / high cohesion)

1. In my opinion `PlagiarismCaseService` should not have a dependency on `org.thymeleaf`. Instead of creating a context and passing it to the mailService, it would be better, if you can pass the `content` and the `notificationUrl`.
2. `PlagiarismCaseService` should also not include a dependency to the `artemisServicerUrl`

**Comment by rriyaldhi** [comment]
> I am not happy with some of the design changes, because you violate best practices for object oriented programming (in particular you do not achieve low coupling / high cohesion)
> 
> 1. In my opinion `PlagiarismCaseService` should not have a dependency on `org.thymeleaf`. Instead of creating a context and passing it to the mailService, it would be better, if you can pass the `content` and the `notificationUrl`.
> 2. `PlagiarismCaseService` should also not include a dependency to the `artemisServicerUrl`

Changes are simplified so that all the comments are addressed.

**Review by frederik-kt — APPROVED** [review]
New subject looks good.

**Review by danielbode — APPROVED** [review]
Retested on ts0. Still works!

**Review by TimOrtel — APPROVED** [review]
In the testing session the email worked correctly and the changes to code are simply and straightforward. Good job!

**Pull request merged**